website/project/metadata/datacite.json

```diff
@@ -0,0 +1,111 @@
+{
+    "_id": "datacite",
+    "name": "Datacite",
+    "schema_version": 1,
+    "category": "file",
+    "schema": {
```

Comment:
Notes

* Look into JSON schema
* Validate metadata records against this schema

See registration metaschemas (validate_metadata) for examples.

Reply:
You have to do some finagling with `not` to handle the non-editable fields:

```python
from jsonschema import validate

# A sample schema, like what we'd get from json.load()
schema = {
    "type": "object",
    "properties" : {
         # NOTE: readOnly is only an annotation--it isn't used for validation
        "title": {"type" : "number", 'readOnly': True},
        "description" : {"type": "string", 'readOnly': True},
        "funder" : {"type" : "string"},
    },
    'not': {
        'anyOf': [
            {'required': ['title']},
            {'required': ['description']}
        ]
    }
}

validate({'funder': 'ljaf'}, schema)  # passes

validate({'funder': 'ljaf', 'title': 'nope'}, schema)  # fails
# jsonschema.exceptions.ValidationError: {'anyOf': [{'required': ['title']}, {'required': ['description']}]} is not allowed for {'funder': 'ljaf', 'title': 'nope'}
#
# Failed validating 'not' in schema:
#     {'not': {'anyOf': [{'required': ['title']},
#                        {'required': ['description']}]},
#      'properties': {'description': {'readOnly': True, 'type': 'string'},
#                     'funder': {'type': 'string'},
#                     'title': {'readOnly': True, 'type': 'number'}},
#      'type': 'object'}
#
# On instance:
#     {'funder': 'ljaf', 'title': 'nope'}
```